index.js

```diff
@@ -0,0 +1,75 @@
+const fp = require('fastify-plugin')
+
+function ignorePath (path, pathsToIgnore) {
+  if (Array.isArray(pathsToIgnore)) {
+    return pathsToIgnore.some((predicate) =>
+      predicate instanceof RegExp
+        ? predicate.test(path)
+        : typeof predicate === 'string'
+          ? predicate === path
+          : false
+    )
+  }
+
+  // If is regex
+  if (pathsToIgnore instanceof RegExp) return pathsToIgnore.test(path)
+
+  // Plain string
+  if (typeof pathsToIgnore === 'string') return pathsToIgnore === path
+}
+
+function onSendHandler (req, reply, payload, done) {
+  // If is a stream, we remove any header without appending
+  // content-length one, and resume it right away
+  if (typeof payload.resume === 'function') {
+    reply.removeHeader('content-type')
+    payload.resume()
+    return done(null, null)
+  }
+
+  const size = '' + Buffer.byteLength(payload)
+
+  reply.header('content-length', size)
+
+  done(null, null)
+}
+
+function getSetupHeadRoutingForGet (fastify, pathsToIgnore) {
+  return function setupHeadRouteForGet (routeOpts) {
+    const {
+      method,
+      path,
+      url,
+      handler,
+      prefix,
+      routePath,
+      ...routeConfig
+    } = routeOpts
+    const isNotGetMethod = method !== 'GET'
+    const shouldIgnore = pathsToIgnore && ignorePath(path, pathsToIgnore)
+
+    // Only GET routes and allowed paths
+    if (isNotGetMethod || shouldIgnore) return
+
+    fastify.route({
+      ...routeConfig,
+      path,
+      handler,
+      method: 'HEAD',
+      onSend: onSendHandler
```

Comment:
You might have an `onSend` in `routeConfig`. If it is present, it should be converted to an array and add as the latest value.

Reply:
True I missed that, thanks!

Reply:
Done, please see [8c2b06f](https://github.com/MetCoder95/fastify-get-head/pull/1/commits/8c2b06f0a081d4ab75c70a1db999f24b6a654797)